src/app/data-model-provider/MetadataTypes.h

```diff
@@ -87,15 +109,85 @@ enum class AttributeQualityFlags : uint32_t
 struct AttributeEntry
 {
     AttributeId attributeId;
-    BitFlags<AttributeQualityFlags> flags;
 
-    // read/write access will be missing if read/write is NOT allowed
-    //
-    // NOTE: this should be compacted for size
-    std::optional<Access::Privilege> readPrivilege;  // generally defaults to View if readable
-    std::optional<Access::Privilege> writePrivilege; // generally defaults to Operate if writable
+    // Constructor
+
+    _StartBitFieldInit; // Disabling '-Wconversion' & '-Wconversion'
+
+    constexpr AttributeEntry(AttributeId id = 0, BitMask<AttributeQualityFlags> attrQualityFlags = BitMask<AttributeQualityFlags>(),
+                             std::optional<Access::Privilege> readPriv  = std::nullopt,
+                             std::optional<Access::Privilege> writePriv = std::nullopt) :
+        attributeId{ id },
+        mask{ attrQualityFlags.Raw() & kAttrQualityMask, (readPriv.has_value() ? to_underlying(*readPriv) : 0) & kPrivilegeMask,
+              (writePriv.has_value() ? to_underlying(*writePriv) : 0) & kPrivilegeMask }
```

Comment:
```suggestion
        mask{ attrQualityFlags.Raw() & kAttrQualityMask, (readPriv.has_value() ? (to_underlying(*readPriv)  & kPrivilegeMask) : 0),
              (writePriv.has_value() ? (to_underlying(*writePriv) & kPrivilegeMask) : 0 }
```

I really wish we had transform available to get more readable code...